src/analysis/return_value.rs

```diff
@@ -120,19 +120,32 @@ pub fn analyze(
 
     let parameter = parameter.as_ref().map(|lib_par| {
         let par = analysis::Parameter::from_return_value(env, lib_par, configured_functions);
-        if let Ok(rust_type) = RustType::builder(env, typ)
-            .direction(par.lib_par.direction)
-            .try_from_glib(&par.try_from_glib)
-            .try_build()
+        if par.lib_par.is_array
+            && !par.lib_par.zero_terminated.unwrap_or(true)
+            && par.lib_par.array_length.is_none()
         {
-            used_types.extend(rust_type.into_used_types());
-        }
+            // The array length for the return value of this function can not be inferred
+            // from any argument. How the length should be inferred is usually specified
+            // in the doc of the function, so there's nothing we can do automatically.
+            // Currently the functions matching this condition are explicitly marked
+            // `manual = true` in the matching Gir.toml.
+            // FIXME: `FixedArray`s could be handled automatically
```

Comment:
Is that hard to add? Also I thought we have some heuristics here somewhere if there's a parameter called `len`, `size` or so

Reply:
It should not be too hard, but that's a different issue from the one this PR is trying to fix imho.
https://github.com/gtk-rs/gir/issues/1655